ObjectTracking/src/bin/basic_control_1d.rs

```diff
@@ -172,8 +174,142 @@ impl<T: ToInputArray> ToInputArrayExt for T {
+}
+
+impl MJpegTracingSubscriber {
+    fn new() -> Self {
+        Self {
+            visitor: std::sync::Mutex::new(MJpegVisitor::new()),
+        }
+    }
+    fn init(self) {
+        let mjpeg_server = self.visitor.lock().unwrap().mjpeg_server.clone();
+        std::thread::spawn(move || mjpeg_server.run("0.0.0.0:8088").unwrap());
+        tracing::dispatcher::set_global_default(self.into()).unwrap();
+    }
+}
+
+impl tracing::Subscriber for MJpegTracingSubscriber {
+    // We toggle interest on and off depending on whether anyone is listening
+    fn register_callsite(&self, metadata: &tracing::Metadata<'_>) -> Interest {
+        Interest::sometimes()
+    }
+
+    // If the mjpeg server is behind, disable subscription for a bit
+    fn enabled(&self, metadata: &tracing::Metadata<'_>) -> bool {
+        match self.visitor.try_lock() {
+            Ok(visitor) => !visitor.mjpeg_server.is_full(),
+            _ => false,
+        }
+    }
+
+    fn new_span(&self, span: &span::Attributes<'_>) -> span::Id {
+        todo!("implement new_span({:?})", span)
+    }
+
+    fn record(&self, span: &span::Id, values: &span::Record<'_>) {
+        todo!("implement record({:?}, {:?})", span, values)
+    }
+
+    fn record_follows_from(&self, span: &span::Id, follows: &span::Id) {
+        todo!("implement record_follows_from({:?}, {:?})", span, follows)
+    }
+
+    fn event(&self, event: &tracing::Event<'_>) {
+        match self.visitor.try_lock() {
+            Ok(ref mut visitor) => event.record(visitor.deref_mut()),
+            // visitor is already behind, so let's skip logging for now.
+            _ => (),
+        }
+    }
+
+    fn enter(&self, span: &span::Id) {
+        todo!("implement enter({:?})", span)
+    }
+
+    fn exit(&self, span: &span::Id) {
+        todo!("implement exit({:?})", span)
+    }
+}
+
+struct MJpegVisitor {
+    mjpeg_server: std::sync::Arc<mjpeg_rs::MJpeg>,
+}
+
+impl MJpegVisitor {
+    fn new() -> Self {
+        Self {
+            mjpeg_server: std::sync::Arc::new(mjpeg_rs::MJpeg::new()),
+        }
+    }
+}
+
+impl tracing::field::Visit for MJpegVisitor {
+    fn record_value(&mut self, field: &tracing::field::Field, value: &dyn valuable::Valuable) {
+        value.visit(self)
+    }
+
+    fn record_debug(&mut self, field: &tracing::field::Field, value: &dyn std::fmt::Debug) {
+        todo!()
+    }
+}
+
+impl valuable::Visit for MJpegVisitor {
+    fn visit_named_fields(&mut self, named_values: &valuable::NamedValues<'_>) {
+        todo!("visit_named_fields({:?})", named_values);
+        let _ = named_values;
+    }
+
+    fn visit_unnamed_fields(&mut self, values: &[valuable::Value<'_>]) {
+        for value in values {
+            if let Some(value) = value.as_listable() {
+                value.visit(self)
+            }
+        }
+    }
+
+    fn visit_primitive_slice(&mut self, slice: valuable::Slice<'_>) {
+        if let valuable::Slice::U8(data) = slice {
```

Comment:
This is a bit questionable: we assume that all `Slice::U8`s are jpegs. Can anyone think of a better way to signal that we want our payload to be interpreted as a jpeg?